### PR TITLE
Constructor computes checksums (so users don't need to initialize the object)

### PR DESCRIPTION
**Pull request by pirapira**
### What this PR does

This PR makes ContractSourceManager's constructor compute the checksums.  Before this PR, users had to initialize the object by calling `compute_checksums()` before calling some other methods.

### Why I'm making this PR

It's nicer when users don't need to care about the internal states of the class.

### What's tricky about this PR (if any)

I made sure that once initialized `contracts_checksums` and `overall_checksum` do not change.

----

Any reviewer can check these:

* [x] If the PR is fixing a bug or adding a feature, add an entry to CHANGELOG.md.
* [x] If the PR changed a Solidity source, run `make compile_contracts` and add the resulting `raiden_contracts/data/contracts.json` in the PR.
* [x] If the PR is changing documentation only, add `[skip ci]` in the commit message so Travis does not waste time.
    * [x] But, if the PR changes comments in a Solidity source, do not add `[skip ci]` and let Travis check the hash of the source.
* [x] In Python, use keyword arguments
* [x] Squash unnecessary commits
* [x] Comment commits
* [x] Follow naming conventions
    * `solidityFunction`
    * `_solidity_argument`
    * `solidity_variable`
    * `python_variable`
    * `PYTHON_CONSTANT`
* [x] Follow the Signature Convention in [CONTRIBUTING.md](./CONTRIBUTING.md)
* For each new contract
    * [x] The deployment script deploys the new contract.
    * [x] `etherscan_verify.py` runs on the new contract.
* Bookkeep
    * [x] The gas cost of new functions are stored in `gas.json`.
* Solidity specific conventions
    * [x] Document arguments of functions in natspec
    * [x] Care reentrancy problems
* [x] When you catch a require() failure in Solidity, look for a specific error message like `pytest.raises(TransactionFailed, match="error message"):`

And before "merge" all checkboxes have to be checked.  If you find redundant points, remove them.

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -21,3 +21,4 @@ isort
 pipdeptree
 eth-typing<2.0.0
 eth-abi<=1.2.2
+six>=1.12
```
Comment (konradkonrad) [inline requirements-dev.txt]: This looks unrelated to the rationale of the PR. Is 2/3 compatibility really a concern?

Comment (pirapira) [reply requirements-dev.txt]: This caused CI failures unrelated to this PR. But, I've added this line on `master` already. This diff will disappear when I rebase.

Comment (pirapira) [reply requirements-dev.txt]: This diff has disappeared.
